### PR TITLE
Add JS-backed string literals

### PR DESCRIPTION
**Pull request by FetBoba**
This PR adds string support to the WASM backend by lowering MLX string literals to JS strings through a host import.

### CODE REVIEW

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala [hunk 1]
```diff
@@ -132,18 +134,36 @@ abstract class WasmDiffMaker extends LlirDiffMaker:
...
       end mkQuery
 
-      val importObj =
-        doc"""{ #{  # "system": { #{  # "mem": new WebAssembly.Memory({initial: 100}) #}  # } #}  # }"""
       val jsStr =
-        doc"""await wasm.binaryenPrintFuncRes( #  #{ `$modWat # `, # $importObj, # exports => exports.${mainFnNme}(), #}  # );"""
+        doc"""
+          await (() => {
+            # const watSrc = $modWatJsLit;
+            # const memMatch = watSrc.match(/\(import\s+"system"\s+"mem"\s+\(memory\s+(\d+)\)\)/);
```
Comment (LPTK) [inline hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: What is this crazy regexp for? Please don't do that.

First of all, could you please explain why you think we need more than one memory here? And what's the strange business with the computation of `minPages`?

Comment (FetBoba) [reply hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: Yes, it was very stupid to do it in this way. Sorry for that. I complicated things too much.
We only need one memory (system.mem), but its initial pages must satisfy the module’s declared minimum. minPages is now computed in codegen as ceil(max(totalStringBytes,1)/65536) and passed directly.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala [hunk 2]
```diff
@@ -287,12 +303,63 @@ class Ctx(
...
+        if newMin =/= existing.minPages then
+          memoryImports(idx) = existing.copy(minPages = newMin)
+      case N =>
+        val idx = memoryImports.size
+        memoryImports += MemoryImport(module, name, minPages)
+        cachedMemoryImport(key) = idx
+
+  /** Returns the minimum page requirement of memory import (`module`, `name`) if present. */
+  def getMemoryImportMinPages(module: Str, name: Str): Opt[Int] =
+    memoryImports.find(m => m.module == module && m.name == name).map(_.minPages)
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]:
```suggestion
    memoryImports.find(m => m.module === module && m.name === name).map(_.minPages)
```

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala [hunk 3]
```diff
@@ -425,6 +494,6 @@ class Ctx(
 
   def toWat: Document =
     val startDef = startFunc.toSeq.map(funcIdx => doc"(start ${funcIdx.toWat})")
-    doc"(module #{  # ${(types.toSeq.map(_.toWat) ++ globals.toSeq.map(_.toWat) ++ startDef ++ funcs.toSeq.map(_.toWat)).mkDocument(doc" # ")}) #} "
+    doc"(module #{  # ${(types.toSeq.map(_.toWat) ++ memoryImports.toSeq.map(_.toWat) ++ functionImports.toSeq.map(_.toWat) ++ dataSegments.toSeq.map(_.toWat) ++ globals.toSeq.map(_.toWat) ++ startDef ++ funcs.toSeq.map(_.toWat)).mkDocument(doc" # ")}) #} "
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: Try to make lines not exceed 120 columns.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 4]
```diff
@@ -200,6 +205,61 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
...
+        nextStringDataOffset = offset + byteLen
+        StringLitInfo(offset = offset, byteLen = byteLen, watBytes = watBytes)
+    )
+
+  /**
+   * Ensures imports required for string materialization exist and returns the constructor function.
+   */
+  private def ensureStringImports(using Ctx): FuncIdx =
+    val minBytes = nextStringDataOffset max 1
+    val minPages = (minBytes + 65535) / 65536
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Please avoid using magic numbers like `65536`. You should bind it to a named Scala constant with an appropriate name. Here, something like `pageSize`, I guess?

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 5]
```diff
@@ -1211,7 +1282,10 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
   def program(p: Program, exprt: Opt[BlockMemberSymbol], wd: io.Path)(using
       Raise,
       Scope
-  ): (Document, Str) =
+  ): (Document, Str, Int) =
+    stringLits.clear()
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: What's this `clear` for?

It's the second time you add this sort of weird statements. Is this generated by AI?

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 6]
```diff
@@ -1211,7 +1282,10 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
...
+    nextStringDataOffset = 0
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Same question.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 7]
```diff
@@ -1231,6 +1305,18 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
 
     val ctx = Ctx.empty
     given Ctx = ctx
+
+    // Check whether there is strigns in block because function imports
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]:
```suggestion
    // Check whether there are strings in the block [because function imports]
```
But "because function imports" doesn't make grammatical sense; please rephrase.

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala [hunk 8]
```diff
@@ -132,18 +134,34 @@ abstract class WasmDiffMaker extends LlirDiffMaker:
...
-          case n if n >= 0 => out.substring(0, n)
-          case _ => ""
+        // Omit the REPL tail only when it is explicitly "undefined" or unit.
+        val lines = out.linesIterator.toList
+        val result =
+          lines.lastOption match
+            case S(last) if last.trim === "undefined" || last.trim === "()" =>
+              lines.dropRight(1).mkString("\n")
+            case _ =>
+              lines.mkString("\n")
```
Comment (LPTK) [inline hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: Can you explain why you made this change?

This code was originally copied from `JSBackendDiffMaker.scala`, which still has the old version. If your version is better, explain why, and share the code between the two so they don't diverge again.

**Review by LPTK — APPROVED** [review]
Otherwise LGTM, but would like to see reviews from others.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 9]
```diff
@@ -10,12 +10,13 @@ import document.*
 import document.Document
 import js.CodeBuilder
 import semantics.*, Elaborator.State
-import syntax.Tree.{BoolLit, IntLit, Ident}
+import syntax.Tree.{BoolLit, IntLit, StrLit, Ident}
 import text.Param as WasmParam
 import Message.MessageContext
 import Scope.scope
+import hkmc2.codegen.BlockTraverser
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Nit: This import is redundant.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala [hunk 10]
```diff
@@ -287,12 +303,63 @@ class Ctx(
...
       namedFuncs(_) = numIdx
     FuncIdx(funcInfo.id.getOrElse(numIdx))
 
+  /**
+   * Adds a function import into this context.
+   *
+   * Returns the function index in the global function index space.
+   */
+  def addFunctionImport(sym: Opt[Symbol], funcImport: FuncImport): FuncIdx =
+    val numIdx = NumIdx(functionImports.size)
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: It seems like this would break if this function is called after `addFunc` is invoked at least once. 

Could we just derive the `NumIdx` of this function import using `functionImports.size + funcs.size` similar to `addFunc`? This would also address this comment: https://github.com/hkust-taco/mlscript/pull/402/changes#r2877874447

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 11]
```diff
@@ -200,6 +205,61 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
...
+    )
+
+  /**
+   * Ensures imports required for string materialization exist and returns the constructor function.
+   */
+  private def ensureStringImports(using Ctx): FuncIdx =
+    val minBytes = nextStringDataOffset max 1
+    val minPages = (minBytes + 65535) / 65536
+    val systemImportMod = "system"
+    val stringFromUtf16ImportNme = "mlx_str_from_utf16"
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: I think these can also be made named constants outside of the function scope. Perhaps store them in an `object ExternIntrinsics`?

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 12]
```diff
@@ -200,6 +205,61 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
...
+        val offset = (nextStringDataOffset + 1) & ~1
+        val byteLen = value.length * 2
+        nextStringDataOffset = offset + byteLen
+        StringLitInfo(offset = offset, byteLen = byteLen, watBytes = watBytes)
+    )
+
+  /**
+   * Ensures imports required for string materialization exist and returns the constructor function.
+   */
+  private def ensureStringImports(using Ctx): FuncIdx =
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Nit: I suggest `getOrLoadStrCtorFunction` - `ensure`-prefix may be confused with assertion-like functions.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala [hunk 13]
```diff
@@ -287,12 +303,63 @@ class Ctx(
...
+
+  /**
+   * Returns the cached function import for (`module`, `name`), creating it with `createImport`
+   * if needed.
+   */
+  def getOrCreateFunctionImport(
+      module: Str,
+      name: Str,
+      createImport: => FuncImport
+  ): FuncIdx =
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: Nit: Rewrite into this form to allow for braceless syntax

```suggestion
  def getOrCreateFunctionImport(
    module: Str,
    name: Str,
  )(createImport: => FuncImport): FuncIdx =
```

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 14]
```diff
@@ -200,6 +205,61 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
...
+            )
+          )
+        )
+        FuncImport(
+          module = systemImportMod,
+          name = stringFromUtf16ImportNme,
+          id = S(SymIdx(stringFromUtf16ImportNme)),
+          typeIdx = importTy
+        )
+    )
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Re https://github.com/hkust-taco/mlscript/pull/402/changes#r2877854309:

```suggestion
    ctx.getOrCreateFunctionImport(
      module = systemImportMod,
      name = stringFromUtf16ImportNme,
    ):
      val importTy = ctx.addType(
        sym = N,
        TypeInfo(
          id = N,
          FunctionType(
            params = Seq(
              WasmParam(N, RefType.anyref),
              WasmParam(N, RefType.anyref)
            ),
            results = Seq(Result(RefType.anyref))
          )
        )
      )
      FuncImport(
        module = systemImportMod,
        name = stringFromUtf16ImportNme,
        id = S(SymIdx(stringFromUtf16ImportNme)),
        typeIdx = importTy
      )
```

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 15]
```diff
@@ -1231,6 +1305,18 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
...
+    // must be declared before function definitions to keep indices stable.
+    var hasStringLiteral = false
+    val traverser = new BlockTraverser:
+      override def applyValue(v: Value): Unit =
+        v match
+          case Value.Lit(StrLit(_)) => hasStringLiteral = true
+          case _ => ()
+    traverser.applyBlock(p.main)
+    if hasStringLiteral then
+      ensureStringImports
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Rather than doing this, I think we should just lazily include the string extern functions as it is used.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Wasm.scala [hunk 16]
```diff
@@ -191,6 +191,26 @@ case class LocalIdx(idx: Index) extends CtxIdx(idx)
...
+    id: Opt[SymIdx],
+    typeIdx: TypeIdx,
+) extends ToWat:
+  def toWat: Document =
+    doc"""(import "${module}" "${name}" (func ${id.fold(doc"")(_.toWat)} (type ${typeIdx.toWat})))"""
+
+/** A data segment entry. */
+case class DataSegment(offset: Int, bytes: Str) extends ToWat:
+  def toWat: Document =
+    doc"""(data (i32.const ${offset}) "${bytes}")"""
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Wasm.scala]: Please do not hardcode the offset like this. The wasm spec explicitly allows an instruction in place of `(i32.const $offset)`, so accept an instruction there instead.

If you want to keep the `offset: Int` overload for convenience, you can have a secondary constructor that specifies a compile-time known offset.

File: hkmc2/shared/src/test/mlscript/wasm/Strings.mls [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+:wasm
+
+:wat
+"Hello World!"
+//│ Wat:
+//│ (module
+//│   (type (func (param (ref null any)) (param (ref null any)) (result (ref null any))))
+//│   (type $Object (sub (struct (field $$tag (mut i32)))))
+//│   (type (func (result (ref null any))))
+//│   (import "system" "mem" (memory 1))
```
Comment (Derppening) [inline hkmc2/shared/src/test/mlscript/wasm/Strings.mls:24]: Can we use `memory 0` by default?

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala [hunk 18]
```diff
@@ -1310,7 +1400,8 @@ class WatBuilder(using TraceLogger, State) extends CodeBuilder:
 
     ctx.addFunc(S(entrySym), entryFnInfo)
 
-    (ctx.toWat, entryNme)
+    val systemMemMinPages = ctx.getMemoryImportMinPages("system", "mem").getOrElse(1)
```
Comment (Derppening) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Nit: I think this can be `0` if no strings are present.

File: hkmc2/shared/src/test/mlscript/wasm/Strings.mls [hunk 19]
```diff
@@ -0,0 +1,90 @@
+
+:global
+:wasm
+
+:wat
+"Hello World!"
```
Comment (Derppening) [inline hkmc2/shared/src/test/mlscript/wasm/Strings.mls]: I think it is also worth adding test cases for empty strings and duplication strings. In particular, the duplicate string test case should show the WAT output to verify that the string deduplication is indeed performed.

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala [hunk 20]
```diff
@@ -66,9 +67,10 @@ abstract class WasmDiffMaker extends LlirDiffMaker:
       val low = ltl.givenIn:
         codegen.Lowering()
       val le = low.program(trm)
-      val (modWat, mainFnNme) = ltl.givenIn:
+      val (modWat, mainFnNme, systemMemMinPages) = ltl.givenIn:
         baseScp.nest.givenIn:
           WatBuilder().program(le, N, wd)
+      val modWatJsLit = Tree.StrLit(modWat.mkString()).idStr
```
Comment (Derppening) [inline hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: I don't think `Tree.StrLit` is supposed to be used this way. Could you explain why this change is needed?

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala [hunk 21]
```diff
@@ -132,18 +134,34 @@ abstract class WasmDiffMaker extends LlirDiffMaker:
...
+            # const watSrc = $modWatJsLit;
+            # const mem = new WebAssembly.Memory({ initial: $systemMemMinPages });
+            # const decodeUtf16 = new TextDecoder("utf-16le");
+            # const imports = {
+                "system": {
+                  "mem": mem,
+                  "mlx_str_from_utf16": (ptr, byteLen) =>
+                    decodeUtf16.decode(new Uint8Array(mem.buffer, ptr, byteLen))
+                }
+              };
```
Comment (Derppening) [inline hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: Nit: Please keep the `importObj` variable and inline the highlighted contents into that `doc`-interpolated string.

File: hkmc2/shared/src/test/mlscript/wasm/Strings.mls (hunk 19 above)
Comment (LPTK) [reply hkmc2/shared/src/test/mlscript/wasm/Strings.mls]: Let's _only_ show the WAT of the duplicated case, to avodi pollution.

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala (hunk 20 above)
Comment (LPTK) [reply hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: If you do want to reuse the escaping behavior, it would be better to refactor it into a utility function. Also see the ocmment on `idStr`.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala (hunk 2 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: Ok done

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 4 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok done

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 9 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok removed

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala (hunk 13 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: Ok done

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 14 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok done

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala (hunk 3 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: Ok changed

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 12 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok renamed

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 7 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: I removed comment and string scan code as was suggested by @Derppening.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala (hunk 10 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Ctx.scala]: Ok I changed it. But I also introduced funcInfosByIndex in Ctx otherwise last test case in String.mls fails with  `Missing function definition for symbol `member:id`` error (which means that indices are not mapped correctly in `getFuncInfo`)

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 15 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok I removed this

File: hkmc2/shared/src/test/mlscript/wasm/Strings.mls (hunk 17 above)
Comment (FetBoba) [reply hkmc2/shared/src/test/mlscript/wasm/Strings.mls:24]: Yes

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 18 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok changed

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Wasm.scala (hunk 16 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/Wasm.scala]: Ok I changed it. I think overload is redundant because it will have only one call site in WatBuilder.scala

File: hkmc2/shared/src/test/mlscript/wasm/Strings.mls (hunk 19 above)
Comment (FetBoba) [reply hkmc2/shared/src/test/mlscript/wasm/Strings.mls]: Ok added these tests and removed other wat flags

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala (hunk 21 above)
Comment (FetBoba) [reply hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: Ok done

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 6 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok removed

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 11 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Ok done

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala (hunk 8 above)
Comment (FetBoba) [reply hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: I changed to old version. I thought that there might be a case where output doesn't have a trailing sentinel line. But turns out that output is always wrapped in `try { ... }`.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala (hunk 5 above)
Comment (FetBoba) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/wasm/text/WatBuilder.scala]: Yes I was using AI to assist me but didn't check this part thoroughly after that (which I usually do). I removed it.

File: hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala (hunk 20 above)
Comment (FetBoba) [reply hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: I changed it to just use `JSBuilder.makeStringLiteral`. Is it ok?

Comment (LPTK) [reply hkmc2DiffTests/src/test/scala/hkmc2/WasmDiffMaker.scala]: I guess it's ok for now.

**Comment by LPTK** [comment]
@FetBoba Please fix the conflicts and then I think we can merge.
